--- toplev.c	2007-04-26 00:46:51.000000000 -0600
+++ toplev.c	2007-11-04 15:55:34.000000000 -0700
@@ -1081,6 +1081,8 @@
   init_cgraph ();
   init_final (main_input_filename);
   coverage_init (aux_base_name);
+  if (flag_syntax_only) 
+    ffi_init (asm_out_file, main_input_filename);
 
   timevar_push (TV_PARSE);
 
@@ -1346,9 +1348,12 @@
 	   , indent, *indent != 0 ? " " : "",
 	   lang_hooks.name, version_string, TARGET_NAME,
 	   indent, __VERSION__);
-  fnotice (file, "%s%sGGC heuristics: --param ggc-min-expand=%d --param ggc-min-heapsize=%d\n",
-	   indent, *indent != 0 ? " " : "",
-	   PARAM_VALUE (GGC_MIN_EXPAND), PARAM_VALUE (GGC_MIN_HEAPSIZE));
+  if (flag_syntax_only)
+    print_ffi_version (file, indent);
+  else
+    fnotice (file, "%s%sGGC heuristics: --param ggc-min-expand=%d --param ggc-min-heapsize=%d\n",
+             indent, *indent != 0 ? " " : "",
+             PARAM_VALUE (GGC_MIN_EXPAND), PARAM_VALUE (GGC_MIN_HEAPSIZE));
 }
 
 /* Print an option value and return the adjusted position in the line.
@@ -2214,6 +2219,8 @@
   input_filename = "<built-in>";
   input_line = 0;
 #endif
+  if (flag_syntax_only)
+    ffi_early_init ();
   if (lang_hooks.init () == 0)
     return 0;
   input_location = save_loc;
