--- objc/objc-act.c.orig	2007-07-03 16:52:55.000000000 -0600
+++ objc/objc-act.c	2007-11-04 16:11:02.000000000 -0700
@@ -4309,7 +4309,9 @@
   /* Suppress outputting debug symbols, because
      dbxout_init hasn'r been called yet.  */
   write_symbols = NO_DEBUG;
-  debug_hooks = &do_nothing_debug_hooks;
+  if (!flag_syntax_only) 
+    debug_hooks = &do_nothing_debug_hooks;
+
 
 #ifdef OBJCPLUS
   push_lang_context (lang_name_c); /* extern "C" */
@@ -16369,6 +16371,19 @@
 	}
     }
   /* APPLE LOCAL end C* property (Radar 4436866) */
+  if (flag_syntax_only) {
+    switch (TREE_CODE (class)) {
+    case CLASS_INTERFACE_TYPE:
+      ffi_rest_of_objc_class_compilation (class);
+      break;
+    case CATEGORY_INTERFACE_TYPE:
+      ffi_rest_of_objc_category_compilation (class);
+      break;
+    case PROTOCOL_INTERFACE_TYPE:
+      ffi_rest_of_objc_protocol_compilation (class);
+      break;
+    }
+  }
 }
 
 static tree
