--- passes.c.orig	2009-09-15 07:45:48.000000000 -0600
+++ passes.c	2009-09-15 07:46:20.000000000 -0600
@@ -187,6 +187,9 @@
   /* Let cgraph know about the existence of variables.  */
   if (TREE_CODE (decl) == VAR_DECL && !DECL_EXTERNAL (decl))
     cgraph_varpool_node (decl);
+
+  if (flag_syntax_only) 
+    ffi_rest_of_decl_compilation (decl, top_level, at_end);
 }
 
 /* Called after finishing a record, union or enumeral type.  */
