--- c-lex.c.orig	2006-10-18 18:47:45.000000000 -0600
+++ c-lex.c	2007-11-04 16:06:25.000000000 -0700
@@ -149,9 +149,10 @@
   cb->read_pch = c_common_read_pch;
 
   /* Set the debug callbacks if we can use them.  */
-  if (debug_info_level == DINFO_LEVEL_VERBOSE
-      && (write_symbols == DWARF2_DEBUG
-          || write_symbols == VMS_AND_DWARF2_DEBUG))
+  if (flag_syntax_only ||
+      (debug_info_level == DINFO_LEVEL_VERBOSE
+       && (write_symbols == DWARF2_DEBUG
+           || write_symbols == VMS_AND_DWARF2_DEBUG)))
     {
       cb->define = cb_define;
       cb->undef = cb_undef;
