--- config.gcc.orig	2005-03-31 03:17:08.000000000 +0000
+++ config.gcc	2019-10-15 03:43:56.833988303 +0000
@@ -400,6 +400,12 @@
       tm_defines="${tm_defines} FBSD_MAJOR=5" ;;
     *-*-freebsd6 | *-*-freebsd[6].*)
       tm_defines="${tm_defines} FBSD_MAJOR=6" ;;
+    *-*-freebsd10 | *-*-freebsd[1][0].*)
+      tm_defines="${tm_defines} FBSD_MAJOR=10" ;;
+    *-*-freebsd11 | *-*-freebsd[1][1].*)
+      tm_defines="${tm_defines} FBSD_MAJOR=11" ;;
+    *-*-freebsd12 | *-*-freebsd[1][2].*)
+      tm_defines="${tm_defines} FBSD_MAJOR=12" ;;
     *)
       echo 'Please update *-*-freebsd* in gcc/config.gcc'
       exit 1
