--- Makefile.in.orig	2007-04-26 00:46:51.000000000 -0600
+++ Makefile.in	2007-11-04 15:52:59.000000000 -0700
@@ -966,7 +966,7 @@
  varasm.o varray.o vec.o version.o vmsdbgout.o xcoffout.o alloc-pool.o	   \
  et-forest.o cfghooks.o bt-load.o pretty-print.o $(GGC) web.o passes.o	   \
  rtl-profile.o tree-profile.o rtlhooks.o cfgexpand.o lambda-mat.o          \
- lambda-trans.o	lambda-code.o tree-loop-linear.o tree-object-size.o
+ lambda-trans.o	lambda-code.o tree-loop-linear.o tree-object-size.o ffi.o
 # APPLE LOCAL end lno, loops-to-memset
 
 OBJS-md = $(out_object_file)
@@ -2020,6 +2020,7 @@
    $(RTL_H) $(FLAGS_H) output.h vmsdbg.h debug.h langhooks.h function.h $(TARGET_H)
 xcoffout.o : xcoffout.c $(CONFIG_H) $(SYSTEM_H) coretypes.h $(TM_H) $(TREE_H) $(RTL_H) \
    xcoffout.h $(FLAGS_H) toplev.h output.h dbxout.h $(GGC_H) $(TARGET_H)
+ffi.o : ffi.c $(CONFIG_H) system.h $(TREE_H) coretypes.h $(TM_H)
 emit-rtl.o : emit-rtl.c $(CONFIG_H) $(SYSTEM_H) coretypes.h $(TM_H) $(RTL_H) $(TREE_H) \
    $(FLAGS_H) function.h $(REGS_H) insn-config.h $(RECOG_H) real.h $(GGC_H) \
    $(EXPR_H) $(OBSTACK_H) hard-reg-set.h bitmap.h toplev.h \
