--- passes.c.orig	2007-06-21 16:29:39.000000000 -0600
+++ passes.c	2007-11-04 16:04:40.000000000 -0700
@@ -255,6 +255,9 @@
   /* Let cgraph know about the existance of variables.  */
   if (TREE_CODE (decl) == VAR_DECL && !DECL_EXTERNAL (decl))
     cgraph_varpool_node (decl);
+
+  if (flag_syntax_only) 
+    ffi_rest_of_decl_compilation (decl, top_level, at_end);
 }
 
 /* Called after finishing a record, union or enumeral type.  */
