--- toplev.c.orig	2010-12-29 03:47:35.128212251 -0700
+++ toplev.c	2010-12-29 03:43:41.728213630 -0700
@@ -983,6 +983,8 @@
   init_cgraph ();
   init_final (main_input_filename);
   coverage_init (aux_base_name);
+  if (flag_syntax_only) 
+    ffi_init (asm_out_file, main_input_filename);
 
   timevar_push (TV_PARSE);
 
@@ -1122,15 +1124,19 @@
 #ifndef __VERSION__
 #define __VERSION__ "[?]"
 #endif
-  fprintf (file,
-	   file == stderr ? _(fmt1) : fmt1,
-	   indent, *indent != 0 ? " " : "",
-	   lang_hooks.name, version_string, TARGET_NAME,
-	   indent, __VERSION__);
-  fprintf (file,
-	   file == stderr ? _(fmt2) : fmt2,
-	   indent, *indent != 0 ? " " : "",
-	   PARAM_VALUE (GGC_MIN_EXPAND), PARAM_VALUE (GGC_MIN_HEAPSIZE));
+  if (flag_syntax_only)
+    print_ffi_version (file, indent);
+  else {
+    fprintf (file,
+             file == stderr ? _(fmt1) : fmt1,
+             indent, *indent != 0 ? " " : "",
+             lang_hooks.name, version_string, TARGET_NAME,
+             indent, __VERSION__);
+    fprintf (file,
+             file == stderr ? _(fmt2) : fmt2,
+             indent, *indent != 0 ? " " : "",
+             PARAM_VALUE (GGC_MIN_EXPAND), PARAM_VALUE (GGC_MIN_HEAPSIZE));
+  }
 }
 
 /* Print an option value and return the adjusted position in the line.
@@ -1840,6 +1846,8 @@
   input_filename = "<built-in>";
   input_line = 0;
 #endif
+  if (flag_syntax_only)
+    ffi_early_init ();
   if (lang_hooks.init () == 0)
     return 0;
   input_location = save_loc;
@@ -1932,6 +1940,12 @@
 
   process_options ();
 
+  
+  if (no_backend && flag_syntax_only) {
+    errorcount++;
+  }
+
+
   /* Don't do any more if an error has already occurred.  */
   if (!errorcount)
     {
