--- c-lex.c.orig	2005-04-04 11:09:26.000000000 -0600
+++ c-lex.c	2005-04-06 00:13:14.000000000 -0600
@@ -105,9 +105,10 @@
   cb->read_pch = c_common_read_pch;
 
   /* Set the debug callbacks if we can use them.  */
-  if (debug_info_level == DINFO_LEVEL_VERBOSE
-      && (write_symbols == DWARF2_DEBUG
-          || write_symbols == VMS_AND_DWARF2_DEBUG))
+  if (flag_syntax_only ||
+      (debug_info_level == DINFO_LEVEL_VERBOSE
+       && (write_symbols == DWARF2_DEBUG
+           || write_symbols == VMS_AND_DWARF2_DEBUG)))
     {
       cb->define = cb_define;
       cb->undef = cb_undef;
